pepdbagent/modules/user.py

```diff
@@ -0,0 +1,196 @@
+import logging
+from typing import Union
+
+from sqlalchemy import and_, delete, select
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import IntegrityError
+
+from pepdbagent.const import (
+    PKG_NAME,
+)
+
+from pepdbagent.db_utils import BaseEngine, User, Stars
+from pepdbagent.modules.project import PEPDatabaseProject
+from pepdbagent.models import AnnotationList, AnnotationModel
+from pepdbagent.exceptions import ProjectNotInFavorites, ProjectAlreadyInFavorites
+
+
+_LOGGER = logging.getLogger(PKG_NAME)
+
+
+class PEPDatabaseUser:
+    """
+    Class that represents Project in Database.
+
+    While using this class, user can create, retrieve, delete, and update projects from database
+    """
+
+    def __init__(self, pep_db_engine: BaseEngine):
+        """
+        :param pep_db_engine: pepdbengine object with sa engine
+        """
+        self._sa_engine = pep_db_engine.engine
+        self._pep_db_engine = pep_db_engine
+
+    def create_user(self, namespace: str) -> int:
+        """
+        Create new user
+
+        :param namespace: user namespace
+        :return: user id
+        """
+        new_user_raw = User(namespace=namespace)
+
+        with Session(self._sa_engine) as session:
+            session.add(new_user_raw)
+            session.commit()
+            user_id = new_user_raw.id
+        return user_id
+
+    def get_user_id(self, namespace: str) -> Union[int, None]:
+        """
+        Get user id using username
+
+        :param namespace: user namespace
+        :return: user id
+        """
+        statement = select(User.id).where(User.namespace == namespace)
+        with Session(self._sa_engine) as session:
+            result = session.execute(statement).one_or_none()
+
+        if result:
+            return result[0]
+        return None
+
+    def add_to_favorites(self, namespace, project_namespace, project_name, project_tag) -> None:
```

Comment:
So the class is `User`. Should this method be `add_project_to_favorites?`. Since it will be called like:
```python
user = User()
user.add_project_to_favorites(...)
```

`user.add_to_favorites` seems less clear to me. Not too strongly opinionated here, though.